src/dex/layouts/TopMenuBar.tsx

```diff
@@ -1,15 +1,28 @@
 import { NavLink } from "react-router-dom";
-import { Box, Flex, Menu, MenuItem } from "@chakra-ui/react";
+import { Box, Flex, Menu, MenuItem, Alert, Text } from "@chakra-ui/react";
 import { useDexContext } from "@dex/hooks";
-import { Color, HashDaoLogo, Text, WalletConnection } from "@shared/ui-kit";
+import { Color, HashDaoLogo, WalletConnection } from "@shared/ui-kit";
 import { isMobile } from "react-device-detect";
+import { useEffect, useState } from "react";
 
 export interface TopMenuBarProps {
   menuOptions: Array<string>;
 }
 
 export function TopMenuBar(props: TopMenuBarProps): JSX.Element {
   const { app, wallet } = useDexContext(({ app, context, wallet }) => ({ app, context, wallet }));
+  const [reconnectionInProgress, setReconnectionInProgress] = useState(false);
+
+  useEffect(() => {
+    if (localStorage.getItem("reconnectionInProgress")) {
+      setReconnectionInProgress(true);
+
+      setTimeout(() => {
+        localStorage.removeItem("reconnectionInProgress");
+        setReconnectionInProgress(false);
+      }, 10000);
```

Comment:
> Is it the flow when user changes the network?
> 
> * we set the reconnect "in progress"
> * page gets reloaded
> * we see the alert about reconnection in progress
> * in 10 seconds the alert is gone
> 
> I see a potential issue:
> 
> * if user switches to the "unsupported" network, in 10 seconds the alert is gone and if user did not read the msg or refreshes the page user still would be "stuck" without any additional information that they need to switch the network.
> 
> Is there a reliable way to read current network and show "you are using unsupported network" on a consistent base?

Nice question, i expected it. To be honest - no, i checked before because was under the same thing. I didn't found the right way to check before switch to one of the unsupported networks.